vault/logical_system_test.go

```diff
@@ -2533,3 +2535,140 @@ func TestSystemBackend_OpenAPI(t *testing.T) {
 		t.Fatalf("expected to find path '/rotate'")
 	}
 }
+
+func testCreateSecret(t *testing.T, core *Core, token, path, key, value string) {
+	t.Helper()
+	req := logical.TestRequest(t, logical.UpdateOperation, path)
+	req.Data[key] = value
+	req.ClientToken = token
+	resp, err := core.HandleRequest(namespace.RootContext(nil), req)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if resp != nil {
+		t.Fatalf("bad: %#v", resp)
+	}
+}
+
+func TestSystemBackend_InternalUIFilteredPath(t *testing.T) {
+	core, b, rootToken := testCoreSystemBackend(t)
+
+	less := func(s1, s2 string) bool { return s1 < s2 }
+	ignoreOrder := []cmp.Option{cmpopts.EquateEmpty(), cmpopts.SortSlices(less)}
```

Comment:
Is this necessary to add? the slices should, i believe, already be sorted from the storage backend

Reply:
That was not my experience.

Reply:
I think it's logical.ListResponse() that does the sorting so from the API perspective we see it sorted, but the backend can return in any order.

Reply:
We're already going through logical.ListResponse, and no, it doesn't sort:

```
func ListResponse(keys []string) *Response {
	resp := &Response{
		Data: map[string]interface{}{},
	}
	if len(keys) != 0 {
		resp.Data["keys"] = keys
	}
	return resp
}
```

Reply:
Most storage backends return list values in sorted order, maybe we should add that to inmem too?

Reply:
I would prefer either the status quo, or that we make it mandatory, e.g. like Jeff was saying having ListResponse do it - or change it for all, make it part of the API, and optionally have ListResponse verify it.  If we just do it for inmem it could lead to more bugs with the backends that don't do that as we (or our users) start taking it more for granted.  I would also prefer that we do that as a separate change rather than burdening this one with it.

Reply:
Sorting at the storage layer would enable us to determine if an additional sort is necessary. For instance consul returns data already sorted, so an additional sort call would be unnecessary. I agree that it isn't required for this PR, but if we go that route we can remove this dependency